content/articles/getting-started-with-unit-testing-with-spring-boot/index.md

```diff
@@ -0,0 +1,125 @@
+
+![Spring Initializer](/engineering-education/getting-started-with-unit-testing-with-spring-boot/spring-initializer.png)
+
+> Note: I have renamed my Package name to `UnitTesting`.
+
+Click next to proceed. Type *web* to search the required dependencies in the search bar. Select *Spring Web* and click *finish* to download the initializr template:
+
+![Spring Initializer](/engineering-education/getting-started-with-unit-testing-with-spring-boot/spring-initializer-dependency.png)
+
+We've prepared the environment successfully.🔥 Now we need to test the controller. In `/src/main/java/UnitTesting` create a new package named `controller`. Proceed to create a Java class named `HelloContoller.java` in  `/src/main/java/UnitTesting/controller`. Within that file, add the code below:
+```java
+package UnitTesting.controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+@RestController
+public class HelloController {
+    @GetMapping("/hello")
+    public String hello(@RequestParam(name = "name", defaultValue = "Hello world") String name){
+        return String.format(name);
+    }
+}
+```
+Let us look at the code snippet above:
+- `@GetMapping` is an annotation used to map the request to the method.
+- `@RequestParam` is an annotation used to map the request parameter to the method parameter.
+- `@RestController` is an annotation used to indicate that the class is a controller.
+
+Let us now generate the controller test.  In the `HelloController.java` file, right-click and select `Generate..>Test..` in the menu. Select the available method (which we want to test) under the member and click ok.
+
+![Spring Initializer](/engineering-education/getting-started-with-unit-testing-with-spring-boot/testing.png)
+
+A new file `/src/main/test/java/UnitTesting/controller/HelloControllerTest.java` will be created. Modify the file to have the code below:
+```java
+package UnitTesting.controller;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+class HelloControllerTest {
+    @Test
+    void hello() {
+        HelloController controller = new HelloController(); // instance of the controller
+        String response = controller.hello("Hello world"); // act
+        Assertions.assertEquals("Hello world", response); // assert
+    }
+}
+```
+So far, we have created a test without including the Spring context. Let us generate another test using the JUnit 5 extension, which will have Spring extensions provided by Spring.  
+
+Repeat the same procedure to generate another test. Name the second test class `HelloControllerIntTest.java`. In `/src/main/test/java/controller/HelloControllerIntTest.java`, modify the code as shown below:
```

Comment:
As I mentioned in another review comment, packages with tests in them should be in `src/test/java` not `src/main/java`